core/beacon/types.go

```diff
@@ -157,34 +138,43 @@ func ExecutableDataToBlock(params ExecutableDataV1) (*types.Block, error) {
 	if params.BaseFeePerGas != nil && (params.BaseFeePerGas.Sign() == -1 || params.BaseFeePerGas.BitLen() > 256) {
 		return nil, fmt.Errorf("invalid baseFeePerGas: %v", params.BaseFeePerGas)
 	}
+	// Only set withdrawalRoot if it is non-nil. This allows CLs to use
+	// ExecutableDataV2 before withdrawals are enabled by marshaling
+	// Withdrawals as the json null value.
+	var withdrawalRoot *common.Hash
+	if params.Withdrawals != nil {
+		h := types.DeriveSha(types.Withdrawals(params.Withdrawals), trie.NewStackTrie(nil))
+		withdrawalRoot = &h
+	}
 	header := &types.Header{
-		ParentHash:  params.ParentHash,
-		UncleHash:   types.EmptyUncleHash,
-		Coinbase:    params.FeeRecipient,
-		Root:        params.StateRoot,
-		TxHash:      types.DeriveSha(types.Transactions(txs), trie.NewStackTrie(nil)),
-		ReceiptHash: params.ReceiptsRoot,
-		Bloom:       types.BytesToBloom(params.LogsBloom),
-		Difficulty:  common.Big0,
-		Number:      new(big.Int).SetUint64(params.Number),
-		GasLimit:    params.GasLimit,
-		GasUsed:     params.GasUsed,
-		Time:        params.Timestamp,
-		BaseFee:     params.BaseFeePerGas,
-		Extra:       params.ExtraData,
-		MixDigest:   params.Random,
+		ParentHash:     params.ParentHash,
+		UncleHash:      types.EmptyUncleHash,
+		Coinbase:       params.FeeRecipient,
+		Root:           params.StateRoot,
+		TxHash:         types.DeriveSha(types.Transactions(txs), trie.NewStackTrie(nil)),
+		ReceiptHash:    params.ReceiptsRoot,
+		Bloom:          types.BytesToBloom(params.LogsBloom),
+		Difficulty:     common.Big0,
+		Number:         new(big.Int).SetUint64(params.Number),
+		GasLimit:       params.GasLimit,
+		GasUsed:        params.GasUsed,
+		Time:           params.Timestamp,
+		BaseFee:        params.BaseFeePerGas,
+		Extra:          params.ExtraData,
+		MixDigest:      params.Random,
+		WithdrawalHash: withdrawalRoot,
```

Comment:
```suggestion
		WithdrawalRoot: withdrawalRoot,
```
maybe ? I don't know ?

Reply:
The reason for this name was to follow along the naming convention of `TxHash` and `ReceiptHash` (which fwiw, I think would be better if with `s/Hash/Root`).